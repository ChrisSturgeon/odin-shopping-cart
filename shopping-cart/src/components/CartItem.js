import '../styles/cartItem.css';

export default function CartItem(props) {
  const path = props.artwork;
  const totalPrice =
    Math.round((props.price * props.quantity + Number.EPSILON) * 100) / 100;

  return (
    <div className="cart-item">
      <img src={path} alt="record artwork"></img>
      <div className="record-details">
        <div>{props.album}</div>
        <div>{props.artist}</div>
      </div>
      <div className="quantity">Quantity: {props.quantity}</div>
      <div>£{props.price} each</div>
      <div className="total-price">£{totalPrice} total</div>
      <button
        onClick={() => {
          console.log(props.sku);
          props.remove(props.sku);
        }}
      >
        <i className="fa-solid fa-trash-can"></i>
      </button>
    </div>
  );
}
