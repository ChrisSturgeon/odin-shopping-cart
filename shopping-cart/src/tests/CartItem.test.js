import React from 'react';
import { render, screen } from '@testing-library/react';
import CartItem from '../components/CartItem';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

describe('CartItem tests', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <CartItem price={17.99} album="Bitches Brew" artist="Miles Davis" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders artist name', () => {
    render(
      <CartItem price={17.99} album="Bitches Brew" artist="Miles Davis" />
    );
    expect(screen.getByText('Miles Davis')).toBeInTheDocument();
  });

  it('renders album name', () => {
    render(
      <CartItem price={17.99} album="Bitches Brew" artist="Miles Davis" />
    );
    expect(screen.getByText('Bitches Brew')).toBeInTheDocument();
  });

  it('renders correct quantity name', () => {
    render(
      <CartItem
        price={17.99}
        album="Bitches Brew"
        artist="Miles Davis"
        quantity={22}
      />
    );
    expect(screen.getByText('Quantity: 22')).toBeInTheDocument();
  });
});
