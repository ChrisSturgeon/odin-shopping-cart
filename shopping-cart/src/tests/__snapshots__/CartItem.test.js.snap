// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="cart-item"
  >
    <img
      alt="record artwork"
    />
    <div
      class="record-details"
    >
      <div>
        Bitches Brew
      </div>
      <div>
        Miles Davis
      </div>
    </div>
    <div
      class="quantity"
    >
      Quantity: 
    </div>
    <div>
      £17.99 each
    </div>
    <div
      class="total-price"
    >
      £NaN total
    </div>
    <button>
      <i
        class="fa-solid fa-trash-can"
      />
    </button>
  </div>
</DocumentFragment>
`;
